openam-authentication/wrenam-auth-duo/src/main/java/org/wrensecurity/wrenam/authentication/modules/duo/Duo.java

```diff
@@ -89,6 +112,10 @@ public int process(Callback[] callbacks, int state) throws AuthLoginException {
                 debug.message("Initializing Cisco Duo authentication for user '" + username + "'.");
                 try {
                     transactionId = duoClient.initAuth(username);
+                    statusRequest = duoClient.getAuthStatus(transactionId);
+                    pollingAssistant.start(statusRequest);
+                    expireTime = System.currentTimeMillis() + pollingTimeout;
+                    return VERIFY_DUO_AUTH_STATUS_STATE;
```

Comment:
I don't think this change is correct as Duo can return success status right away. You don't want to run through the the whole authentication round-trip in that case. The original logic was better in this sense.

Reply:
See my comment below.